src/Processors/Formats/Impl/PrettyBlockOutputFormat.cpp

```diff
@@ -33,6 +33,10 @@ void PrettyBlockOutputFormat::calculateWidths(
     WidthsPerColumn & widths, Widths & max_padded_widths, Widths & name_widths)
 {
     size_t num_rows = std::min(chunk.getNumRows(), format_settings.pretty.max_rows);
+
+    auto max_row_number = std::to_string(num_rows);
```

Comment:
I'd rather use `log10` instead of creating temporary string.
There is no difference in performance in this case (one operation per block of data), but it will be just better.

Reply:
Ok, I will fix it.